src/Microsoft.ServiceFabric.Actors/Runtime/ActorStateProviderHelper.cs

```diff
@@ -496,6 +496,99 @@ internal static IActorStateProvider CreateDefaultStateProvider(ActorTypeInformat
             return Task.FromResult(actorQueryResult);
         }
 
+        internal Task<PagedResult<ActorId>> GetStoredActorIdsForKvsAsync(
+            int itemsCount,
+            ContinuationToken continuationToken,
+            KeyValueStoreReplica replica,
+            CancellationToken cancellationToken)
+        {
+            var actorIdList = new List<ActorId>();
+            var actorQueryResult = new PagedResult<ActorId>();
+
+            IEnumerator<KeyValueStoreItem> enumerator;
+            bool enumHasMoreEntries;
+            using var txn = replica.CreateTransaction();
+
+            // Find continuation point for enumeration
+            if (continuationToken != null)
+            {
+                long previousActorCount = 0L;
+                if (long.TryParse((string)continuationToken.Marker, out previousActorCount))
+                {
+                    enumerator = replica.Enumerate(txn, ActorStateProviderHelper.ActorPresenceStorageKeyPrefix, true);
+                    enumHasMoreEntries = this.GetContinuationPointByActorCount(previousActorCount, enumerator, cancellationToken);
+                }
+                else
+                {
+                    string lastSeenActorStorageKey = continuationToken.Marker.ToString();
+                    enumerator = replica.Enumerate(txn, lastSeenActorStorageKey, false);
+                    enumHasMoreEntries = enumerator.MoveNext();
+                    if (enumHasMoreEntries)
+                    {
+                        enumHasMoreEntries = enumerator.MoveNext();
+                    }
+                }
+
+                if (!enumHasMoreEntries)
+                {
+                    // We are here means the current snapshot that enumerator represents
+                    // has less entries that what ContinuationToken contains.
+                    return Task.FromResult(actorQueryResult);
+                }
+            }
+            else
+            {
+                enumerator = replica.Enumerate(txn, ActorStateProviderHelper.ActorPresenceStorageKeyPrefix, true);
+                enumHasMoreEntries = enumerator.MoveNext();
+            }
+
+            if (!enumHasMoreEntries)
+            {
+                return Task.FromResult(actorQueryResult);
+            }
+
+            while (enumHasMoreEntries)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var storageKey = enumerator.Current.Metadata.Key;
+                var actorId = GetActorIdFromPresenceStorageKey(storageKey);
+
+                if (actorId != null)
+                {
+                    actorIdList.Add(actorId);

```

Comment:
I thought strictPrefix:false starts the enumerator form the first element that either matches or greater than the prefix and we need to check if the Current.MetaData.Key startswith presencekey tag(@@)?

Reply:
Updated